### PR TITLE
AZP: Lock resources to prevent double start

### PR DESCRIPTION
**Pull request by Alexey-Rivkin**
## What
Prevent results tampering due to parallel start on 2+ PRs.

### CODE REVIEW

File: buildlib/azure-pipelines-perf.yml [hunk 1]
```diff
@@ -31,6 +33,28 @@ stages:
           - ucx_perf
 
         steps:
+          - bash: |
+              set -xeE
+              lock_file="$(SharedDir)/on-air.lock"
```
Comment (artemry-nv) [inline buildlib/azure-pipelines-perf.yml:38]: Aren't there any standard GHA mechanisms to prevent parallel workflow runs?
CC @B-a-S

Comment (Alexey-Rivkin) [reply buildlib/azure-pipelines-perf.yml:38]: Nope. First thing I checked.

**Comment by Alexey-Rivkin** [comment]
> @Alexey-Rivkin which tests fail without this PR?

No test fails. The purpose of this PR is to prevent concurrent tests for different PRs, skewing final results.

**Comment by yosefe** [comment]
> > @Alexey-Rivkin which tests fail without this PR?
> 
> No test fails. The purpose of this PR is to prevent concurrent tests for different PRs, skewing final results.

Can't we limit number of agents to 1?

**Comment by Alexey-Rivkin** [comment]
> > > @Alexey-Rivkin which tests fail without this PR?
> > 
> > 
> > No test fails. The purpose of this PR is to prevent concurrent tests for different PRs, skewing final results.
> 
> Can't we limit number of agents to 1?

Good point - done that.
I would still prefer to have this locking logic for added safety.

**Comment by yosefe** [comment]
> > > > @Alexey-Rivkin which tests fail without this PR?
> > > 
> > > 
> > > No test fails. The purpose of this PR is to prevent concurrent tests for different PRs, skewing final results.
> > 
> > 
> > Can't we limit number of agents to 1?
> 
> Good point - done that. I would still prefer to have this locking logic for added safety.

If not needed, I'd prefer to not add more logic to the testing infra

**Comment by Alexey-Rivkin** [comment]
Required as a prerequisite to https://github.com/openucx/ucx/pull/9431

**Pull request closed without merging**